sns_ik_examples/src/ik_tests.cpp

```diff
@@ -44,6 +44,25 @@ double fRand(double min, double max)
   return min + f * (max - min);
 }
 
+double getDeltaWithLimits(double value, double desired_delta,
+                          double limit_min, double limit_max)
+{
+  double valid_delta;
+  double lower_delta = value-desired_delta;
+  double upper_delta = value+desired_delta;
+  double upper_side = (double)rand() / RAND_MAX >= 0.5;
+  if( upper_side && upper_delta > limit_max ){
+      valid_delta = lower_delta;
+  } else if( upper_side && upper_delta <= limit_max) {
```

Comment:
mmm i wrote this out, and I think it might be less clear:
```
  if( upper_side ) {
    if( upper_delta > limit_max ) {
      valid_delta = lower_delta;
    } else {
      valid_delta = upper_delta;
    }
  } else {
    if( lower_delta < limit_min ){
      valid_delta = upper_delta;
    } else {
      valid_delta = lower_delta;
    }
  }
```
It should be right around the same level of efficiency.